bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/config/config.component.ts

```diff
@@ -0,0 +1,139 @@
+import { AccessPolicyService } from "../../shared/access-policies/access-policy.service";
+
+class ServiceAccountConfig {
+  organizationId: string;
+  serviceAccountId: string;
+  identityUrl: string;
+  apiUrl: string;
+  projects: ProjectListView[];
+}
+
+@Component({
+  selector: "sm-service-account-config",
+  templateUrl: "./config.component.html",
+})
+export class ServiceAccountConfigComponent implements OnInit, OnDestroy {
+  identityUrl: string;
+  apiUrl: string;
+  organizationId: string;
+  serviceAccountId: string;
+  projects: ProjectListView[];
+  hasProjects = false;
+
+  private destroy$ = new Subject<void>();
+  loading = true;
+
+  constructor(
+    private environmentService: EnvironmentService,
+    private route: ActivatedRoute,
+    private platformUtilsService: PlatformUtilsService,
+    private toastService: ToastService,
+    private i18nService: I18nService,
+    private projectService: ProjectService,
+    private accessPolicyService: AccessPolicyService,
+  ) {}
+
+  async ngOnInit() {
+    this.route.params
+      .pipe(
+        concatMap(async (params: Params) => {
+          return await this.load(params.organizationId, params.serviceAccountId);
+        }),
+        takeUntil(this.destroy$),
+      )
+      .subscribe((smConfig) => {
+        this.identityUrl = smConfig.identityUrl;
+        this.apiUrl = smConfig.apiUrl;
+        this.organizationId = smConfig.organizationId;
+        this.serviceAccountId = smConfig.serviceAccountId;
+        this.projects = smConfig.projects;
+
+        this.hasProjects = smConfig.projects.length > 0;
+        this.loading = false;
+      });
+  }
+
+  async load(organizationId: string, serviceAccountId: string): Promise<ServiceAccountConfig> {
+    const environment = await this.environmentService.getEnvironment();
+
+    const allProjects = await this.projectService.getProjects(organizationId);
+    const policies = await this.accessPolicyService.getServiceAccountGrantedPolicies(
+      organizationId,
+      serviceAccountId,
+    );
+
+    const ids = policies.grantedProjectPolicies.map(
+      (policy) => policy.accessPolicy.grantedProjectId,
+    );
+
+    const projects = allProjects.filter((project) =>
+      ids.some((projectId) => projectId === project.id),
+    );
+
+    // The machine account may have access to projects that the user does not have access to.
+    // We show these on the Project tab. so add these projects to this list too.
+    if (allProjects.length < policies.grantedProjectPolicies.length) {
+      for (const policy of policies.grantedProjectPolicies) {
+        if (!projects.some((project) => project.id === policy.accessPolicy.grantedProjectId)) {
+          projects.push({
+            id: policy.accessPolicy.grantedProjectId,
+            name: policy.accessPolicy.grantedProjectName,
+            organizationId: organizationId,
+            linkable: false,
+          } as ProjectListView);
+        }
+      }
+    }
```

Comment:
With this change, the `const projects` is now identical to `policies.grantedProjectPolicies` - so it makes sense to change it to be build from `policies.grantedProjectPolicies` and not from `allProjects`. That way you don't need to filter the list and the only thing that needs to be mapped is `linkable`, which can only be true, if the `projectId` is in `allProjects`.
```suggestion
    const projects = policies.grantedProjectPolicies.map((policy) => {
        return {
            id: policy.accessPolicy.grantedProjectId,
            name: policy.accessPolicy.grantedProjectName,
            organizationId: organizationId,
            linkable: allProjects.some((project) => project.id === policy.accessPolicy.grantedProjectId),
          } as ProjectListView;
    });
```